android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -105,6 +106,131 @@ fun getProductInfo(
     }
 }
 
+fun purchase(
+    activity: Activity?,
+    options: Map<String, Any?>,
```

Comment:
As I mentioned, the main downside of this is that we lose any type of compile time safety, in exchange of unifying the logic of parsing the parameters to PHC. Lmk if you have any concerns over the approach!